ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -617,6 +620,10 @@ void fillChunks() {
           channelReadBuf = chunkFillerChannel.getNextByteBuf(0);
         }
         if (channelReadBuf != null) {
+          // If first chunk is awaiting resolution, and there are more bytes in the channel prepare it first as a data chunk.
+          if (channelReadBuf.readableBytes() > 0 && firstPutChunkAwaitingResolution && !firstPutChunkPrepared) {
+            prepareFirstChunk(false);
+          }
```

Comment:
we can do 
```
if (firstChunk == chunkToFill && firstPutChunkAwaitForResolution) {
    prepareFirstChunk(false);
}
```
In this case, if the current chunk is still the first chunk, meaning first chunk is not in encryption or network, and the first chunk is waiting for resolution, we now know that the resolution is here.